libsolidity/AST.cpp

```diff
@@ -209,6 +209,40 @@ vector<pair<FixedHash<4>, FunctionTypePointer>> const& ContractDefinition::getIn
 	return *m_interfaceFunctionList;
 }
 
+vector<ASTPointer<Declaration>> const& ContractDefinition::getInheritableMembers() const
+{
+	if (!m_inheritableMembers)
+	{
+		set<string> memberSeen;
+		m_inheritableMembers.reset(new vector<ASTPointer<Declaration>>());
+		for (ContractDefinition const* contract: getLinearizedBaseContracts())
+		{
+			for (ASTPointer<FunctionDefinition> const& f: contract->getDefinedFunctions())
+				if (f->isPublic() && !f->isConstructor() && !f->getName().empty()
+					&& memberSeen.count(f->getName()) == 0 && f->isVisibleInDerivedContracts())
+				{
+					memberSeen.insert(f->getName());
+					m_inheritableMembers->push_back(f);
+				}
+
+			for (ASTPointer<VariableDeclaration> const& v: contract->getStateVariables())
+				if (v->isPublic() && memberSeen.count(v->getName()) == 0)
```

Comment:
"Internal" would also suffice. Can you check whether we can make "isVisibleInDerivedContracts" virtual so that it behaves exactly as these conditions here? Then we can treat all the elements of those vectors equally after a cast to a Declaration.